### PR TITLE
Query and update animals table

### PR DESCRIPTION
**Pull request by Kakalanp**
- Adding new queries.
- Tweaked general database structure.
- Adding transactions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,17 +1,73 @@
...
 
-SELECT name, escape_attempts FROM animals WHERE weight_kg >  10.5;
+UPDATE animals SET species = 'unspecified';
+SELECT * FROM animals;
 
-SELECT * FROM animals WHERE neutered = 1;
+ROLLBACK;
 
-SELECT * FROM animals WHERE name != 'Gabumon';
+SELECT * FROM animals;
```
Comment (mikethreels) [inline queries.sql]: - [x] in this transaction you should only check the species column in the animals table.

File: queries.sql [hunk 2]
```diff
@@ -1,17 +1,73 @@
...
+--SECOND TRANSACTION
+
+BEGIN;
+
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+
+COMMIT;
+
+SELECT * FROM animals;
```
Comment (mikethreels) [inline queries.sql]: - [x] in this transaction you only need to check the species table after the commit.

File: queries.sql [hunk 3]
```diff
@@ -1,17 +1,73 @@
...
+
+--THIRD TRANSACTION
+
+BEGIN;
+
+DELETE FROM animals;
```
Comment (mikethreels) [inline queries.sql]: - [x]  if you're deleting the entire table, you should put the asterisk in the query.

File: queries.sql [hunk 4]
```diff
@@ -1,17 +1,73 @@
...
+ROLLBACK TO SP1;
+
+UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0;
+
+COMMIT;
+
+SELECT * FROM animals;
+
+--How many animals are there?
+SELECT COUNT() FROM animals;
```
Comment (mikethreels) [inline queries.sql]: - [x] you should include the asterisk here to make it work.

File: queries.sql [hunk 5]
```diff
@@ -1,17 +1,73 @@
...
+--How many animals have never tried to escape?
+SELECT COUNT() FROM animals WHERE escape_attempts <= 0;
```
Comment (mikethreels) [inline queries.sql]: - [x] you should include the asterisk here to make it work.
- [x] escape_attempts will most likely not be less than 0 therefore you can just put it like `escape_attempts = 0`

File: queries.sql [hunk 6]
```diff
@@ -1,17 +1,73 @@
...
+--What is the average weight of animals?
+SELECT AVG(weight_kg) FROM animals;
+--Who escapes the most, neutered or not neutered animals?
+SELECT MAX(escape_attempts) FROM animals WHERE neutered = true;
```
Comment (mikethreels) [inline queries.sql]: - [x] you should select column neutered first and then group by neutered.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Jaoquin, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Kakalanp) [reply queries.sql]: Done, thanks!

File: queries.sql (hunk 2 above)
Comment (Kakalanp) [reply queries.sql]: Understood, thanks!

File: queries.sql (hunk 3 above)
Comment (Kakalanp) [reply queries.sql]: It works for me without an asterisk, either way, I'll add it, thank you.

File: queries.sql (hunk 5 above)
Comment (Kakalanp) [reply queries.sql]: completely overlooked the asterisk, thanks!

And yeah, it makes sense to keep it to 0, thank you, once more.

File: queries.sql (hunk 6 above)
Comment (Kakalanp) [reply queries.sql]: Right away!

**Comment by Kakalanp** [comment]
Merging the project as it is shown approved in the dashboard

**Pull request merged**